### ISSUE TITLE
Not possible to join two streams with complex key type

### ISSUE THREAD

**Issue by lichutin-st** [LGouellec/kafka-streams-dotnet#71]
Description
===========
Hello! Looks like it's impossible to join two streams with complex key types.

I tried to do so and got an exception
`[StreamsException] The serializer is not compatible to the actual key (Key type: Program.UserKey). Change the default Serdes in StreamConfig or provide correct Serdes via method parameters(using the DSL)`

I debugged through Streamiz.Kafka.Net library and found this:
https://github.com/LGouellec/kafka-streams-dotnet/blob/develop/core/Stream/StreamJoinProps.cs#L221
Looks like Key and Value SerDes are ignored when `StreamJoinProps` is being built from another `StreamJoinProps` (and that operations is performed during the `Join` node building

Nuget version: 1.1.3

How to reproduce
================

```
class UserKey
{
    public string Id { get; set; }
    public string CustomerId { get; set; }
}

IKStream<UserKey, object> stream1;
IKStream<UserKey, object> stream2;

stream1.Join(
    stream2,
    (entry1, entry) => new { entry1, entry },
    JoinWindowOptions.Of(TimeSpan.FromMinutes(1)),
    StreamJoinProps.With(
        keySerde: new JsonSerializer<UserKey>(),
        valueSerde: new JsonSerializer<object>(),
        otherValueSerde: new JsonSerializer<object>()
    )
)
```

Checklist
=========

Please provide the following information:
 - [x] A complete (i.e. we can run it), minimal program demonstrating the problem. No need to supply a project file.
 - [x] A code snippet with your topology builder (ex: builder.Stream<string, string>("topic").to("an-another-topic");)
 - [x] Streamiz.Kafka.Net nuget version.
 - [ ] Apache Kafka version.
 - [ ] Client configuration.
 - [ ] Operating system.
 - [ ] Provide logs (with in debug mode (log4net and StreamConfig.Debug) as necessary in configuration).
 - [ ] Critical issue.

**Comment by LGouellec** [comment]
Hi @lichutin-st

Thanks for your issue. 
Exactly, your analysis is correct. I'll try to fix a new release 1.1.5 ASAP.

Regards,

**Comment by LGouellec** [comment]
Added unit test to specify this use case [see](https://github.com/LGouellec/kafka-streams-dotnet/blob/5632fe82246178b669c9de6b4cec0f77d7bfaf21/test/Streamiz.Kafka.Net.Tests/Processors/KStreamKStreamJoinTests.cs#L308)
